lib/graphql/cache/marshal.rb

```diff
@@ -50,12 +50,25 @@ def read(config, force: false, &block)
       # @param config [Hash] The middleware resolution config hash
       def write(config)
         resolved = yield
+
         document = Deconstructor[resolved].perform
 
-        cache.write(key, document, expires_in: expiry(config))
+        with_resolved_document(document) do |resolved_document|
+          cache.write(key, resolved_document, expires_in: expiry(config))
```

Comment:
Hmmm, I found some issues with my first pass, but not this one.

AR relations should be cacheable--afaik my change shouldn't have caused any new problems around AR relations.

Reply:
Perhaps i'm not doing the right thing. I

 can also convert my ActiveRecord::Relation query `.to_a` in my resolver but before i set up the cache i did not have any errors on my GraphQL queries.

Reply:
@bricesanchez Are you having this issue on this feature branch or on the released gem version?

Reply:
I have this issue on this feature branch. 

I have also cache key collision on nested queries, i've end up with this solution in my API:

```ruby
# lib/graphql/custom_cache_key.rb:1
module Graphql
  class CustomCacheKey
    def initialize(obj, ctx)
      @obj = obj
      @ctx = ctx
    end

    def generate
      [
        @obj.cache_key,
        "current_user:#{@ctx[:current_user].id}"
      ].join('-')
    end
  end
end
```

```ruby
# app/graphql/types/clinic.rb

[...]

  field :mandates, resolver: Resolvers::Clinics::Mandates,
        cache: {
          key: -> (obj, ctx) { ::Graphql::CustomCacheKey.new(obj.mandates, ctx).generate }
        }
[...]
```

Reply:
I have this bug on the feature branch not on the release gem.

Reply:
That's odd! The AR relation, when it hits `Deconstructor#perform` (https://github.com/vincedevendra/graphql-cache/blob/50c74836b46f50d4cfa5cac7c32745cf8b9d6202/lib/graphql/cache/deconstructor.rb#L47-L57), will have a method of `relation`, which means it will fall into `Deconstructor#deconstruct_object`, and that method isn't touched by this branch.